### PR TITLE
Support parsing numeric timestamps since epoch

### PR DESCRIPTION
**Pull request by jachris**

### CODE REVIEW

File: libtenzir/src/series_builder.cpp [hunk 1]
```diff
@@ -1161,70 +1161,93 @@ auto builder_ref::try_atom(detail::atom_view value) -> caf::expected<void> {
...
+                      || std::same_as<FromType, double_type>) {
+          auto since_epoch
+            = cast_value(FromType{}, value, duration_type{}, *unit);
+          if (not since_epoch) {
+            return since_epoch.error();
           }
-        } else {
-          result = cast_value(FromType{}, value, ty);
+          return time{} + *since_epoch;
         }
```
Comment (jachris) [inline libtenzir/src/series_builder.cpp:1224]: This should be the only relevant change, the rest if refactoring.

File: libtenzir/src/series_builder.cpp [hunk 2]
```diff
@@ -1161,70 +1161,92 @@ auto builder_ref::try_atom(detail::atom_view value) -> caf::expected<void> {
...
+        auto unit = full_ty.attribute("since_epoch_unit");
+        if (unit) {
+          auto since_epoch
+            = cast_value(FromType{}, value, duration_type{}, *unit);
+          if (not since_epoch) {
+            return since_epoch.error();
           }
-        } else {
-          result = cast_value(FromType{}, value, ty);
+          return time{} + *since_epoch;
```
Comment (jachris) [inline libtenzir/src/series_builder.cpp:1222]: This should be the only relevant change, the rest is refactoring.

File: libtenzir/src/series_builder.cpp [hunk 3]
```diff
@@ -1161,70 +1161,92 @@ auto builder_ref::try_atom(detail::atom_view value) -> caf::expected<void> {
...
-          if (not result) {
-            result
-              = cast_value(FromType{}, fmt::format("{} {}", value, unit), ty);
+        return result;
+      }
+    } else if constexpr (std::same_as<ToType, time_type>) {
+      if constexpr (std::same_as<FromType, int64_type>
+                    || std::same_as<FromType, uint64_type>
+                    || std::same_as<FromType, double_type>) {
+        auto unit = full_ty.attribute("since_epoch_unit");
```
Comment (jachris) [inline libtenzir/src/series_builder.cpp:1215]: Consider renaming this to `unit` for brevity.

File: libtenzir/src/series_builder.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libtenzir/src/series_builder.cpp:1222]: Thanks, that made the review a whole lot easier.

File: libtenzir/src/series_builder.cpp (hunk 3 above)
Comment (dominiklohmann) [reply libtenzir/src/series_builder.cpp:1215]: I'm in favor of this change.

Comment (dominiklohmann) [inline libtenzir/src/series_builder.cpp:1215]: Can you either document this change or write a follow-up story to document it (and also #unit for timestamps)?

**Review by dominiklohmann — APPROVED** [review]

File: libtenzir/src/series_builder.cpp (hunk 3 above)
Comment (dominiklohmann) [reply libtenzir/src/series_builder.cpp:1215]: At the very least this needs a changelog entry within this PR.

Comment (jachris) [reply libtenzir/src/series_builder.cpp]: A changelog entry was already part of this PR, maybe you missed it. I added documentation for `#unit` in 3cfdcd9e0e895457e70783e2832051697acf9496. Please resolve if you're happy with both.

Comment (dominiklohmann) [reply libtenzir/src/series_builder.cpp]: I did miss it. Thank you!

**Pull request merged**